sdl_android/src/main/java/com/smartdevicelink/api/SdlManager.java

```diff
@@ -0,0 +1,516 @@
+		 */
+		public Builder setAppName(@NonNull final String appName){
+			sdlManager.appName = appName;
+			return this;
+		}
+
+		/**
+		 * Sets the Short Application Name
+		 * @param shortAppName
+		 */
+		public Builder setShortAppName(final String shortAppName) {
+			sdlManager.shortAppName = shortAppName;
+			return this;
+		}
+
+		/**
+		 * Sets the Language of the App
+		 * @param hmiLanguage
+		 */
+		public Builder setLanguage(final Language hmiLanguage){
+			sdlManager.hmiLanguage = hmiLanguage;
+			return this;
+		}
+
+		/**
+		 * Sets the TemplateColorScheme for daytime
+		 * @param dayColorScheme
+		 */
+		public Builder setDayColorScheme(final TemplateColorScheme dayColorScheme){
+			sdlManager.dayColorScheme = dayColorScheme;
+			return this;
+		}
+
+		/**
+		 * Sets the TemplateColorScheme for nighttime
+		 * @param nightColorScheme
+		 */
+		public Builder setnightColorScheme(final TemplateColorScheme nightColorScheme){
+			sdlManager.nightColorScheme = nightColorScheme;
+			return this;
+		}
+
+		/*
+		public Builder setLockScreenConfig (final LockScreenConfig lockScreenConfig){
+			sdlManager.lockScreenConfig = lockScreenConfig;
+			return this;
+		}*/
+
+		/**
+		 * Sets the vector of AppHMIType
+		 * @param hmiTypes
+		 */
+		public Builder setAppTypes(final Vector<AppHMIType> hmiTypes){
+
+			if (hmiTypes != null) {
+				sdlManager.hmiTypes = hmiTypes;
+				sdlManager.isMediaApp = hmiTypes.contains(AppHMIType.MEDIA);
+			}else {
+				Vector<AppHMIType> hmiTypesDefault = new Vector<>();
+				hmiTypesDefault.add(AppHMIType.DEFAULT);
+				sdlManager.hmiTypes = hmiTypesDefault;
+				sdlManager.isMediaApp = false;
+			}
+
+			return this;
+		}
+
+		/**
+		 * Sets the vector of vrSynonyms
+		 * @param vrSynonyms
+		 */
+		public Builder setVrSynonyms(final Vector<String> vrSynonyms) {
+			sdlManager.vrSynonyms = vrSynonyms;
+			return this;
+		}
+
+		/**
+		 * Sets the TTS Name
+		 * @param ttsChunks
+		 */
+		public Builder setTtsName(final Vector<TTSChunk> ttsChunks) {
+			sdlManager.ttsChunks = ttsChunks;
+			return this;
+		}
+
+		/**
+		 * This Object type may change with the transport refactor
+		 * Sets the BaseTransportConfig
+		 * @param transport
+		 */
+		public Builder setTransportType(BaseTransportConfig transport){
+			sdlManager.transport = transport;
+			return this;
+		}
+
+		/**
+		 * Sets the Context
+		 * @param context
+		 */
+		public Builder setContext(Context context){
+			sdlManager.context = context;
+			return this;
+		}
+
+		@SuppressWarnings("unchecked")
+		public SdlManager build() {
+			try {
+				sdlManager.initialize();
+
+				if (sdlManager.hmiTypes == null) {
+					Vector<AppHMIType> hmiTypesDefault = new Vector<>();
+					hmiTypesDefault.add(AppHMIType.DEFAULT);
+					sdlManager.hmiTypes = hmiTypesDefault;
+					sdlManager.isMediaApp = false;
+				}
+
+				if (sdlManager.hmiLanguage == null){
+					sdlManager.hmiLanguage = Language.EN_US;
+				}
+
+				if (sdlManager.appName == null) {
```

Comment:
This check and the `appId` check should be first. If they aren't included there is no reason to continue.